### DISCUSSION TITLE
Access is denied alert after successful login using OIDC

### DISCUSSION THREAD

**Discussion by anmtan** [apache/airflow#21358] [General]
**Apache Airflow version**: 2.0.1

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): 1.19

**Environment**: Kubernetes Executor, Okta OIDC 

- **Cloud provider or hardware configuration**: EKS 1.19, RDS Postrgres DB 
- **OS** (e.g. from /etc/os-release): Centos inside docker
- **Kernel** (e.g. `uname -a`): 
- **Install tools**: helm 
- **Others**: using official docker image apache/airflow:2.0.1-python3.7,  flask-oidc, fab-oidc2

**What happened**:
We have integrated airflow 2 with Okta OIDC authentication/authorization using flask-oidc and fab-oidc2. When user login to Airflow console, user will be redirected to Okta login. Once user successful login to Okta, it will redirect back to airflow. On the airflow home page, there is an alert "Access is denied". If you refresh the page, the Alert will disappear. Even though the alert says access is denied, but the user can do everything. Self registration is enabled, all users have admin privileges. I've checked the user table and role table, both looks right to me and user should have admin privilege. The alert is just a false alarm and it only appears at user login. 

**What you expected to happen**: User login successful without alert

**How to reproduce it**: Airflow 2.0.1 Integrate with Okta OIDC, login to airflow console. After user successfully login to airflow, the landing page will have "Access is denied".

**Anything else we need to know**:

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by gcalmettes** [comment]
Hitting the same behavior after successful login via OIDC
![image](https://user-images.githubusercontent.com/1042512/119667327-7b3c6880-be36-11eb-8b98-06650b121496.png)

**Comment by hao-zhang-aurora** [comment]
We are having the same issue with Airflow 2.1 and Okta. Is there any updates on this?

**Comment by sinhapiyushkr** [comment]
we are having same issue with Airflow 2.1.1 and Google Oauth2.

**Comment by potiuk** [comment]
I can confirm this is the same for me for latest version of Airflow with Google Oauth2. I am not too familiar with the FAB authentication flows and what could be the reason, but maybe someone who knows a bit better @thesuperzapper @kaxil @dpgaspar could help ? I have now working configuration with Google OAuth2 that I can use to quickly test any solution (and happy to implement a fix)  - but I need some guidance on it  and possibly brainstorming where it could come from?

I saw this happening when I enabled - self-registration via Google Oauth as an Admin user. I think the oauth flow somehow puts the "Access Denied" message when you first try to connect and have no auth cookies set yet/never connected and instead of just redirecting to oauth server to perform authentication, the "Access Denied" message is immediately added to the error list.

**Comment by Shivamtayal99** [comment]
We are also facing similar issue in airflow 1.10.14 with OpenID connect.

**Comment by schwartzmx** [comment]
Hitting the same issue in airflow 2.1.2 using Google OAuth2.

**Comment by thesuperzapper** [comment]
Can someone please set `AIRFLOW__LOGGING__FAB_LOGGING_LEVEL = DEBUG`, and provide logs from the webserver when this error happens?

Also, please provide a (sanitized) version of your `webserver_config.py`.

**Comment by schwartzmx** [comment]
Thanks for the reply,  attaching requested info @thesuperzapper.

`webserver_config.py`
```python
import os
import logging

from flask_appbuilder.security.manager import AUTH_OAUTH

basedir = os.path.abspath(os.path.dirname(__file__))
logger = logging.getLogger(__name__)

GOOGLE_CLIENT_ID = os.getenv('GOOGLE_OAUTH2_CLIENT_ID')
GOOGLE_SECRET = os.getenv('GOOGLE_OAUTH2_SECRET')
ADMIN_EMAILS = os.getenv("WEB_ADMIN_EMAILS", "").split(',')

AUTH_TYPE = AUTH_OAUTH
AUTH_ROLE_ADMIN = 'Admin'
AUTH_USER_REGISTRATION = True
AUTH_USER_REGISTRATION_ROLE = 'Admin'  # Testing with just everyone as admin

OAUTH_PROVIDERS = [{
'name':'google',
    'token_key':'access_token',
    'icon':'fa-google',
        'remote_app': {
            'api_base_url':'https://www.googleapis.com/oauth2/v2/',
            'client_kwargs':{
                'scope': 'email profile'
            },
            'access_token_url':'https://accounts.google.com/o/oauth2/token',
            'authorize_url':'https://accounts.google.com/o/oauth2/auth',
            'request_token_url': None,
            'client_id': GOOGLE_CLIENT_ID,
            'client_secret': GOOGLE_SECRET,
        }
}]

# Custom Security Manager in order to get around the `role_keys` missing from Google OAuth response
# See: https://github.com/apache/airflow/issues/16783
from airflow.www.security import AirflowSecurityManager
AUTH_ROLES_MAPPING = {
    "devs": ["Viewer"],
    "admins": ["Admin"]
}

class GoogleAirflowSecurityManager(AirflowSecurityManager):
    def oauth_user_info(self, provider, resp):
        assert provider == "google", "Google provider is only supported in this Security Manager"
        me = self.appbuilder.sm.oauth_remotes[provider].get("userinfo")
        data = me.json()
        email = data.get("email", "")
        # Maps back to AUTH_ROLES_MAPPING keys
        role_keys = ["admins"] if email in ADMIN_EMAILS else ["devs"]
        return {
            "username": "google_" + data.get("id", ""),
            "first_name": data.get("given_name", ""),
            "last_name": data.get("family_name", ""),
            "email": email,
            "role_keys": role_keys
        }

SECURITY_MANAGER_CLASS = GoogleAirflowSecurityManager
```
`webserver.log`
Logs right as the login process occurs and directly after as the home page is rendered
```
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:29 +0000] "GET /airflow/ HTTP/1.1" 302 233 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:29 +0000] "GET /airflow/home HTTP/1.1" 302 341 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | [2021-08-17 05:46:29,307] {views.py:645} DEBUG - Provider: None
airflow-webserver_1  | [2021-08-17 05:46:29,307] {views.py:661} DEBUG - Going to call authorize for: google
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:29 +0000] "GET /airflow/login/?next=http%3A%2F%2Flocalhost%3A8080%2Fairflow%2Fhome HTTP/1.1" 302 1029 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | [2021-08-17 05:46:32,693] {views.py:694} DEBUG - Authorized init
airflow-webserver_1  | 127.0.0.1 - - [17/Aug/2021:05:46:33 +0000] "GET /airflow/health HTTP/1.1" 200 187 "-" "curl/7.64.0"
airflow-webserver_1  | [2021-08-17 05:46:33,530] {views.py:699} DEBUG - OAUTH Authorized resp: {'access_token': '<redacted>', 'expires_in': 3599, 'scope': 'https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/userinfo.profile openid', 'token_type': 'Bearer', 'id_token': '<redacted>', 'expires_at': 1629182792}
airflow-webserver_1  | [2021-08-17 05:46:34,371] {views.py:708} DEBUG - User info retrieved from google: {'username': '<redacted>', 'first_name': '<redacted>', 'last_name': '<redacted>', 'email': '<redacted>', 'role_keys': ['admins']}
airflow-webserver_1  | [2021-08-17 05:46:34,371] {views.py:721} DEBUG - No whitelist for OAuth provider
airflow-webserver_1  | [2021-08-17 05:46:34,378] {manager.py:227} INFO - Updated user <redacted>
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/oauth-authorized/google?state=<redacted>&code=<redacted>&scope=email%20profile%20openid%20https://www.googleapis.com/auth/userinfo.profile%20https://www.googleapis.com/auth/userinfo.email&authuser=0&hd=<redacted>&prompt=none HTTP/1.1" 302 275 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/home HTTP/1.1" 200 57650 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/css/bootstrap.min.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/css/font-awesome.min.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/datepicker/bootstrap-datepicker.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/select2/select2.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/css/flags/flags16.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/css/ab.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/airflowDefaultTheme.3e8bda71892b61b62f94.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/materialIcons.3221294eb511f43d1b15.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/main.e52cf607b64cdcd15089.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/loadingDots.4033edd9abf2750d6f8f.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/bootstrap-datetimepicker.min.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/switch.e97750fdb7423f33656a.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/dags.6c090f6b27d152c78e7a.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/flash.d205b61edc54ed448412.css HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/js/jquery-latest.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/js/ab_filters.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/js/ab_actions.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/css/bootstrap.min.css.map HTTP/1.1" 404 567 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/js/bootstrap.min.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/select2/select2.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/datepicker/bootstrap-datepicker.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/appbuilder/js/ab.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/moment.c1933ee062e9650051f7.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/main.e52cf607b64cdcd15089.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:34 +0000] "GET /airflow/static/dist/bootstrap-datetimepicker.min.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:35 +0000] "GET /airflow/static/dist/bootstrap3-typeahead.min.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:35 +0000] "GET /airflow/static/dist/d3.min.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:35 +0000] "GET /airflow/static/dist/dags.6c090f6b27d152c78e7a.js HTTP/1.1" 304 0 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:35 +0000] "POST /airflow/last_dagruns HTTP/1.1" 200 753 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:35 +0000] "POST /airflow/blocked HTTP/1.1" 200 127 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:35 +0000] "POST /airflow/task_stats HTTP/1.1" 200 6723 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
airflow-webserver_1  | <ip> - - [17/Aug/2021:05:46:35 +0000] "POST /airflow/dag_stats HTTP/1.1" 200 1804 "http://localhost:8080/airflow/home" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36"
```
Additional info:
airflow version: `2.1.2`
python: `python3.8`
airflow.cfg base_url
```
[webserver]
base_url = http://localhost:8080/airflow
```
Google OAuth2 Dashboard configured URIs:
```
Authorized JavaScript origins:
http://localhost:8080

Authorized redirect URIs:
http://localhost:8080/airflow/oauth-authorized/google
```

As the @anmtan said originally,
>The alert is just a false alarm and it only appears at user login.

Refresh of the page removes the banner and the user can carry on,  but it's just something that is odd and would need to be noted to bring up to users.
<img width="1087" alt="image" src="https://user-images.githubusercontent.com/6064408/129672436-d80eed14-c945-4f1c-ae60-0a14c46de6d0.png">

**Comment by JustMaris** [comment]
#12237 seems related

**Comment by yakimetsd** [comment]
The same issue is still present in v2.1.3 coupled to Azure AD OAuth2

**Comment by iercan** [comment]
We have same issue with 2.1.4. I'm also not able to log out from airflow. It redirects to login if I try to logout.

**Comment by omoussa1** [comment]
Hey, has a solution been reached for this?

**Comment by agamjainTW** [comment]
Same issue with  Airflow 2.2.2 and Github Oauth.

**Comment by emilianomoscato** [comment]
Hello, it was not happening in our environment with Airflow 2.0.2 and chart 8.5.3, but after upgrading to Airflow 2.1.4 we start getting this error. We didn't change our Google OAUTH configuration.

**Comment by potiuk** [comment]
Look at changes to Flask App Builder between the two versions. Google Oauth integration and configuration is provided by Flask App Builder, it's not something that AIrflow does. 

* FAB was 3.1.1 in 2.0.1 https://github.com/apache/airflow/blob/constraints-2.0.1/constraints-3.8.txt#L4
* FAB was 3.3.3 in 2.1.4 https://github.com/apache/airflow/blob/constraints-2.1.4/constraints-3.8.txt#L16

You can also open an issue in FAB's issue tracker or discussion there - maybe you will get help https://github.com/dpgaspar/flask-appbuilder/

I am converting it into discussion in case there will be some solutions you would like to post here for others.

**Comment by potiuk** [comment]
BTW. Possibly you can also downgrade your FAB to 3.1.1 as a workaround (until the problem is diagnosed and fixed) - you really want to update to later versions of FAB as they usually contain some security fixes.

**Comment by thesuperzapper** [comment]
A related issue for some people is that they set `AUTH_USER_REGISTRATION_ROLE = "Public"`, and if a user hasn't also been given the `"Viewer"` or `"User"` roles (for example, by using `AUTH_ROLES_MAPPING`), they will always be redirected back to login, even if their password is correct.

NOTE: this is because the `"Public"` role is not actually able to see the Airflow UI.

**Reply by potiuk** [reply]
Yeah. I do recall similar symptoms and the fix that @thesuperzapper  describes sound very close to what is in my (a little faded) memory

**Reply by emilianomoscato** [reply]
Innitially we had this problem with role "Public", changing default to "Viewer" solved it.
But this is a different issue that became after upgrading with community helm chart to Airflow 2.1.4.

I'll come back if we find any workaround.
Thanks!

**Comment by iercan** [comment]
This issue seems resolved on version 2.2.5.

**Reply by SrihithaReddy13** [reply]
I'm using a custom security manager and still see the same error with v 2.3.2 - anyone else?

**Reply by pdylanross** [reply]
I was dealing with this on an airflow 2.1.4 setup (FAB 3.3.3), and I upgraded just FAB to the version set in airflow 2.2's constraints (FAB 3.4.5). This resolved the issue altogether for me. I'm still testing to see if that broke anything else though.

**Reply by brsolomon-deloitte** [reply]
> I'm using a custom security manager and still see the same error with v 2.3.2 - anyone else?

Same here - broken with Airflow 2.3.3.

**Reply by potiuk** [reply]
I guess if some of those who comment here will open an issue about it with logs, configuration and detailed info how to reproduce it, they will increase the probability someone will take a look at it/ commenting that you have the same issue, adds no value to anyone that would like to help (as opposed to raising a reproducible issue).

**Reply by brsolomon-deloitte** [reply]
@potiuk nope - actually, when someone _asserts that the issue has been resolved on a specific Airflow version_, its necessary to reply to dispute and invalidate their claim when the bug is still present on a later version. Plenty of reproduction details have already been posted earlier in the thread.

**Reply by potiuk** [reply]
Or create a new issue. That also works and this is our approach. Yes it is more effort for reporter, but allows to organise work by maintainers better. If you are enough empathetic to understand that you have one issue to handle and maintainers and other people who might help with diagnosing the issue (who BTW spend their nights and weekend volunteering and working for free so that those who report the issue might use the  software for FREE) 

It is also about probability, if you really want someone to take a look at the issue and fix it, do everything to make it easier for that someone. As mentioned above - there are no guarantees that issues reported will be solved. It takes someone of the 2500 volunteers here to acknowledge, diagnose, spend time on fixing and testing an issue to be fixed. If it is not easy, the likelihood no-one will do it is high, but if reporters spend their time on making it easier for anyone to understand and diagnose the isssue, the probability of things being fixed is higher.

So I am just telling you, the easier you make for those who will be (possibly) fixing it, the bigger chance is that it will get, in fact fixed.

(And BTW. I am not the one who is going to fix it, as I have very little knowledge about these parts - it's more of a helpful comment of maintainer who suggests what actions can be taken by those who want to get it fixed to increase probablity of it.

**Comment by dadonnelly316** [comment]
Flask-OIDC should be marked as abandoned and isn't supported by airflow.

This module will internally direct you to the airflow home page after successfully authenticating with the 3rd party identity provider, and then airflow redirects you to the login endpoint since you're not yet authenticated in airflow. Only then are you authenticated with airflow.

The fix would be to redirect yourself to the login endpoint after authenticating with the 3rd party provider

**Comment by thesuperzapper** [comment]
I have a suspicion that at least some people are encountering the `redirect_uri` issue described in https://github.com/apache/airflow/discussions/21850, which is where Airflow does not know it's running under HTTPS because you are using a proxy that provides its own TLS (like an AWS NLB on Kubernetes).

In these cases, you might be able to get the OpenID/OAuth2 integration to work by doing some combination of the following:

- Setting [`AIRFLOW__WEBSERVER__BASE_URL`](https://airflow.apache.org/docs/apache-airflow/2.7.2/configurations-ref.html#base-url) to your domain with the `https://` part, for example: `https://airflow.example.com`
- Setting [`AIRFLOW__WEBSERVER__ENABLE_PROXY_FIX`](https://airflow.apache.org/docs/apache-airflow/2.7.2/configurations-ref.html#enable-proxy-fix)  to `true` (read more about this [here](https://airflow.apache.org/docs/apache-airflow/stable/howto/run-behind-proxy.html))
- Ensuring you configure BOTH the HTTP and HTTPS as redirect URI in OIDC provider, e.g.
    - `https://airflow.example.com/oauth-authorized/okta`
    - `http://airflow.example.com/oauth-authorized/okta`
